alibi_detect/cd/lsdd_online.py

```diff
@@ -0,0 +1,134 @@
+import logging
+import numpy as np
+from typing import Callable, Dict, Optional, Union
+from alibi_detect.utils.frameworks import has_pytorch, has_tensorflow
+
+if has_pytorch:
+    from alibi_detect.cd.pytorch.lsdd_online import LSDDDriftOnlineTorch
+
+if has_tensorflow:
+    from alibi_detect.cd.tensorflow.lsdd_online import LSDDDriftOnlineTF
+
+logger = logging.getLogger(__name__)
+
+
+class LSDDDriftOnline:
+    def __init__(
+            self,
+            x_ref: np.ndarray,
+            ert: float,
+            window_size: int,
+            backend: str = 'tensorflow',
+            preprocess_x_ref: bool = True,
+            preprocess_fn: Optional[Callable] = None,
+            sigma: Optional[np.ndarray] = None,
+            n_bootstraps: int = 1000,
+            n_kernel_centers: Optional[int] = None,
+            lambda_rd_max: float = 0.2,
+            device: Optional[str] = None,
+            input_shape: Optional[tuple] = None,
+            data_type: Optional[str] = None
+    ) -> None:
+        """
+        Online least squares density difference (LSDD) data drift detector using preconfigured thresholds.
+        Motivated by Bu et al. (2017): https://ieeexplore.ieee.org/abstract/document/7890493
+        We have made modifications such that a desired ERT can be accurately targeted however.
+
+        Parameters
+        ----------
+        x_ref
+            Data used as reference distribution.
+        ert
+            The expected run-time (ERT) in the absence of drift.
+        window_size
+            The size of the sliding test-window used to compute the test-statistic.
+            Smaller windows focus on responding quickly to severe drift, larger windows focus on
+            ability to detect slight drift.
+        backend
+            Backend used for the LSDD implementation and configuration.
+        preprocess_x_ref
+            Whether to already preprocess and store the reference data.
+        preprocess_fn
+            Function to preprocess the data before computing the data drift metrics.s
+        sigma
```

Comment:
Again make clear that if `sigma` is not specified, it is inferred from the reference data.